stdlib/format.mli

```diff
@@ -1417,11 +1417,14 @@ val eprintf : ('a, formatter, unit) format -> 'a
 *)
 
 val sprintf : ('a, unit, string) format -> 'a
-(** Same as [printf] above, but instead of printing on a formatter,
+(** Same as {!printf} above, but instead of printing on a formatter,
   returns a string containing the result of formatting the arguments.
   Note that the pretty-printer queue is flushed at the end of {e each
-  call} to [sprintf]. Note that if your format string contains a [%a],
-  you should use [asprintf].
+  call} to [sprintf]. Note that you should most likely use {!asprintf}, which
+  has better support for [%a]. [sprintf] is compatible with [%a] conversions,
+  but it expects values with a different type than when using [%a] with
+  {!printf} or {!fprintf}, where {!asprintf} is compatible with the other [Format]
+  functions. [sprintf] is kept only for backward compatibility.
```

Comment:
I would propose to merge and simplify the first two sentences:

> Note that you likely want to use {!asprintf} which can reuse %a-printer defined for {!fprintf}.
Contrarily, sprintf requires to redefine new %a-printers, and is kept only for backward compatility.